### PR TITLE
On auth expire soon event

### PR DESCRIPTION
**Pull request by JohannesDoberer**

### CODE REVIEW

File: docs/authorization-events.md [hunk 1]
```diff
@@ -31,3 +32,4 @@ You can disable the default behavior of `onAuthExpired` and `onAuthError` by mak
     Return `false` to prevent redirecting to `logoutUrl` after executing this function. It goes to the Luigi main route `/` instead.
 -   `onAuthExpired` **[function](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Statements/function)** is executed if the token expires during runtime, or if Luigi is opened with outdated authorization data in the local storage. Return `false` to prevent redirecting to `logoutUrl` after executing this function.
 -   `onLogout` **[function](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Statements/function)** is executed after the user logs out.
+- `onAuthExpireSoon`**[function](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Statements/function)** is executed before token will expire. For oAuth2 and OIDC it is possible to specify the seconds when the event will be fired by setting **accessTokenExpiringNotificationTime**.
```
Comment (bszwarc) [inline docs/authorization-events.md]:
```suggestion
- `onAuthExpireSoon`**[function](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Statements/function)** is executed before the token expires. For oAuth2 and OIDC you can set the **accessTokenExpiringNotificationTime** to specify the number of seconds required to pass before the notification is fired.
```

File: docs/authorization-configuration.md [hunk 2]
```diff
@@ -78,6 +80,8 @@ auth: {
 - **nonceFn** provides a function that returns a string in order to override the default **nonce**.
 - **logoutFn** provides the function to override the **logoutUrl** functionality for a custom logout. It needs to execute the **logoutCallback()** function after logout.
 - **userInfoFn** provides a function to get user information. It returns a promise of a **userinfo** object which can contain **name**, **email** and **picture** (value is a URL to the image). **Name** or **email** are displayed in the profile drop-down menu and the user’s profile picture is displayed in the top navigation.
+- **accessTokenExpiringNotificationTime** number of seconds before an access token is to expire to raise the **onAuthExpireSoon** event. Default is 300 seconds.
```
Comment (bszwarc) [inline docs/authorization-configuration.md]:
```suggestion
- **accessTokenExpiringNotificationTime** number of seconds that pass before an access token expires and the **onAuthExpireSoon** event is fired. The default value is `300` seconds.
```

File: docs/authorization-configuration.md [hunk 3]
```diff
@@ -78,6 +80,8 @@ auth: {
...
+- **expirationCheckInterval** number of seconds how often the check happens, when token is going to expire. Default is 5 seconds.
```
Comment (bszwarc) [inline docs/authorization-configuration.md]:
```suggestion
- **expirationCheckInterval** the number of seconds to pass between each check if the token is about to expire. The default value is `5` seconds.
```

File: core/examples/luigi-sample-angular/src/luigi-config/extended/auth.js [hunk 4]
```diff
@@ -104,6 +106,13 @@ class Auth {
     onAuthError: (settings, err) => {
       console.log('authErrorHandler 1', err, settings);
       // return false; // prevent redirect to logoutUrl, but go to /
+    },
+    onAuthExpireSoon: settings => {
+      console.log('onAuthExpireSoon ', settings);
+      Luigi.showAlert({
```
Comment (jesusreal) [inline core/examples/luigi-sample-angular/src/luigi-config/extended/auth.js]: Should the warning be displayed by Luigi itself and configurable through Luigi Config, and not configured by each application using Luigi?

Comment (jesusreal) [reply core/examples/luigi-sample-angular/src/luigi-config/extended/auth.js]: Clarified with @hardl, we are enabling already every Luigi application to easily provide custom handling for this event, no further actions are required.

File: docs/authorization-events.md [hunk 5]
```diff
@@ -31,3 +32,5 @@ You can disable the default behavior of `onAuthExpired` and `onAuthError` by mak
     Return `false` to prevent redirecting to `logoutUrl` after executing this function. It goes to the Luigi main route `/` instead.
 -   `onAuthExpired` **[function](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Statements/function)** is executed if the token expires during runtime, or if Luigi is opened with outdated authorization data in the local storage. Return `false` to prevent redirecting to `logoutUrl` after executing this function.
 -   `onLogout` **[function](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Statements/function)** is executed after the user logs out.
+- `onAuthExpireSoon` **[function](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Statements/function)** is executed before the token expires. For oAuth2 and OIDC you can set the **accessTokenExpiringNotificationTime** to specify the number of seconds required to pass before the event is fired. The default value is `60` seconds.
+    - by using oAuth2ImplicitGrant you can specify **expirationCheckInterval**, which are the number of seconds to pass between each check if the token is about to expire. The default value is `5` seconds.
```
Comment (marynaKhromova) [inline docs/authorization-events.md]:
```suggestion
    - by using oAuth2ImplicitGrant you can specify **expirationCheckInterval**, which is the number of seconds to pass between each check if the token is about to expire. The default value is `5` seconds.
```

File: docs/authorization-events.md [hunk 6]
```diff
@@ -31,3 +32,5 @@ You can disable the default behavior of `onAuthExpired` and `onAuthError` by mak
     Return `false` to prevent redirecting to `logoutUrl` after executing this function. It goes to the Luigi main route `/` instead.
 -   `onAuthExpired` **[function](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Statements/function)** is executed if the token expires during runtime, or if Luigi is opened with outdated authorization data in the local storage. Return `false` to prevent redirecting to `logoutUrl` after executing this function.
 -   `onLogout` **[function](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Statements/function)** is executed after the user logs out.
+- `onAuthExpireSoon` **[function](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Statements/function)** is executed before the token expires. For oAuth2 and OIDC you can set the **accessTokenExpiringNotificationTime** to specify the number of seconds required to pass before the event is fired. The default value is `60` seconds.
```
Comment (bszwarc) [inline docs/authorization-events.md]:
```suggestion
- `onAuthExpireSoon` **[function](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Statements/function)** is executed before the token expires. For OAuth2 and OIDC you can set the **accessTokenExpiringNotificationTime** to specify the number of seconds required to pass before the event is fired. The default value is `60` seconds.
```

File: docs/authorization-events.md [hunk 7]
```diff
@@ -31,3 +32,5 @@ You can disable the default behavior of `onAuthExpired` and `onAuthError` by mak
...
+    - by using oAuth2ImplicitGrant you can specify **expirationCheckInterval**, which is the number of seconds to pass between each check if the token is about to expire. The default value is `5` seconds.
```
Comment (bszwarc) [inline docs/authorization-events.md]:
```suggestion
    - by using oAuth2ImplicitGrant you can specify **expirationCheckInterval** which is the number of seconds to pass between each check if the token is about to expire. The default value is `5` seconds.
```

**Pull request merged**